### PR TITLE
Enable MacOS builds

### LINKED ISSUE #11672: Building torchvision for osx-64
### Actual Behavior

```conda create -n pytorch pytorch torchvision``` does not work on osx-64, as the package `torchvision` is not available for this platform.

### Expected Behavior

It would be nice if that worked as expected on osx-64.

### Steps to Reproduce

```conda create -n pytorch pytorch torchvision``` on an osx-64 platform.

##### Anaconda or Miniconda version:

Miniconda for python-3.7, version 4.8.2

##### `conda info`

<details>

```
     active environment : None
            shell level : 0
       user config file : /Users/andre/.condarc
 populated config files : /Users/andre/conda/condarc
          conda version : 4.8.2
    conda-build version : 3.18.11
         python version : 3.7.4.final.0
       virtual packages : __osx=10.15.3
       base environment : /Users/andre/conda  (writable)
           channel URLs : https://www.idiap.ch/software/bob/conda/label/beta/osx-64
                          https://www.idiap.ch/software/bob/conda/label/beta/noarch
                          https://www.idiap.ch/software/bob/conda/osx-64
                          https://www.idiap.ch/software/bob/conda/noarch
                          https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /Users/andre/conda/pkgs
                          /Users/andre/.conda/pkgs
       envs directories : /Users/andre/conda/envs
                          /Users/andre/.conda/envs
               platform : osx-64
             user-agent : conda/4.8.2 requests/2.23.0 CPython/3.7.4 Darwin/19.3.0 OSX/10.15.3
                UID:GID : 501:20
             netrc file : None
           offline mode : False
```

</details>

##### `conda list --show-channel-urls`

<details>

```
# packages in environment at /Users/andre/conda:
#
# Name                    Version                   Build  Channel
alabaster                 0.7.12                   py37_0    defaults
anaconda-client           1.7.2                    py37_0    defaults
appdirs                   1.4.3            py37h28b3542_0    defaults
asn1crypto                1.3.0                    py37_0    defaults
attrs                     19.3.0                     py_0    defaults
babel                     2.8.0                      py_0    defaults
backports                 1.0                        py_2    defaults
backports.functools_lru_cache 1.6.1                      py_0    defaults
backports.tempfile        1.0                        py_1    defaults
backports.weakref         1.0.post1                  py_1    defaults
beautifulsoup4            4.8.2                    py37_0    defaults
black                     19.10b0                    py_0    defaults
bleach                    3.1.0                    py37_0    defaults
bob.devtools              3.1.4b0          py37h661208a_1    https://www.idiap.ch/software/bob/conda/label/beta
bzip2                     1.0.8                h1de35cc_0    defaults
ca-certificates           2020.1.1                      0    defaults
cairo                     1.14.12              hc4e6be7_4    defaults
certifi                   2019.11.28               py37_0    defaults
cffi                      1.14.0           py37hb5b8e2f_0    defaults
chardet                   3.0.4                 py37_1003    defaults
click                     7.0                      py37_0    defaults
click-plugins             1.1.1                      py_0    defaults
clyent                    1.2.2                    py37_1    defaults
cmarkgfm                  0.4.2            py37h1de35cc_0    defaults
conda                     4.8.2                    py37_0    defaults
conda-build               3.18.11                  py37_0    defaults
conda-package-handling    1.6.0            py37h1de35cc_0    defaults
conda-verify              3.4.2                      py_1    defaults
cryptography              2.8              py37ha12b0ac_0    defaults
decorator                 4.4.2                      py_0    defaults
docformatter              1.3.1                      py_0    http://bobconda.lab.idiap.ch/public/conda
docutils                  0.16                     py37_0    defaults
expat                     2.2.6                h0a44026_0    defaults
filelock                  3.0.12                     py_0    defaults
fontconfig                2.13.0               h5d5b041_1    defaults
freetype                  2.9.1                hb4e5f40_0    defaults
fribidi                   1.0.5                h1de35cc_0    defaults
future                    0.18.2                   py37_0    defaults
gettext                   0.19.8.1             h15daf44_3    defaults
gitdb2                    2.0.6                      py_0    defaults
gitpython                 3.0.5                      py_0    defaults
glib                      2.63.1               hd977a24_0    defaults
glob2                     0.7                        py_0    defaults
graphite2                 1.3.13               h2098e52_0    defaults
graphviz                  2.40.1               hefbbd9a_2    defaults
harfbuzz                  1.8.8                hb8d4a28_0    defaults
icu                       58.2                 h4b95b61_1    defaults
idna                      2.8                      py37_0    defaults
imagesize                 1.2.0                      py_0    defaults
importlib_metadata        0.8                      py37_0    https://www.idiap.ch/software/bob/conda
ipython_genutils          0.2.0                    py37_0    defaults
jinja2                    2.11.1                     py_0    defaults
jpeg                      9b                   he5867d9_2    defaults
jsonschema                3.2.0                    py37_0    defaults
jupyter_core              4.6.1                    py37_0    defaults
libarchive                3.3.3                h786848e_5    defaults
libcxx                    4.0.1                hcfea43d_1    defaults
libcxxabi                 4.0.1                hcfea43d_1    defaults
libedit                   3.1.20181209         hb402a30_0    defaults
libffi                    3.2.1                h475c297_4    defaults
libiconv                  1.15                 hdd342a3_7    defaults
liblief                   0.9.0                h2a1bed3_2    defaults
libpng                    1.6.37               ha441bb4_0    defaults
libtiff                   4.1.0                hcb84e12_0    defaults
libxml2                   2.9.9                hf6e021a_1    defaults
libxslt                   1.1.33               h33a18ac_0    defaults
lxml                      4.5.0            py37hef8c89e_0    defaults
lz4-c                     1.8.1.2              h1de35cc_0    defaults
lzo                       2.10                 h362108e_2    defaults
markupsafe                1.1.1            py37h1de35cc_0    defaults
mypy_extensions           0.4.3                    py37_0    defaults
nbformat                  5.0.4                      py_0    defaults
ncurses                   6.2                  h0a44026_0    defaults
openssl                   1.1.1d               h1de35cc_4    defaults
packaging                 19.2                       py_0    defaults
pango                     1.42.4               h060686c_0    defaults
pathspec                  0.7.0                      py_0    defaults
pcre                      8.43                 h0a44026_0    defaults
pip                       20.0.2                   py37_1    defaults
pixman                    0.38.0               h1de35cc_0    defaults
pkginfo                   1.5.0.1                  py37_0    defaults
psutil                    5.7.0            py37h1de35cc_0    defaults
py-lief                   0.9.0            py37h1413db1_2    defaults
pycosat                   0.6.3            py37h1de35cc_0    defaults
pycparser                 2.19                     py37_0    defaults
pygments                  2.5.2                      py_0    defaults
pyopenssl                 19.1.0                   py37_0    defaults
pyparsing                 2.4.6                      py_0    defaults
pyrsistent                0.15.7           py37h1de35cc_0    defaults
pysocks                   1.7.1                    py37_0    defaults
python                    3.7.4                h359304d_1    defaults
python-dateutil           2.8.1                      py_0    defaults
python-gitlab             1.15.0                   py37_0    http://bobconda.lab.idiap.ch/public/conda
python-graphviz           0.13.2                     py_0    defaults
python-libarchive-c       2.8                     py37_13    defaults
pytz                      2019.3                     py_0    defaults
pyyaml                    5.3              py37h1de35cc_0    defaults
readline                  7.0                  h1de35cc_5    defaults
readme_renderer           24.0                     py37_0    defaults
regex                     2020.2.20        py37h1de35cc_0    defaults
requests                  2.23.0                   py37_0    defaults
requests-toolbelt         0.9.1                      py_0    defaults
ripgrep                   11.0.2               he32d670_0    defaults
ruamel_yaml               0.15.87          py37h1de35cc_0    defaults
setuptools                46.0.0                   py37_0    defaults
six                       1.14.0                   py37_0    defaults
smmap2                    2.0.5                    py37_0    defaults
snowballstemmer           2.0.0                      py_0    defaults
soupsieve                 1.9.5                    py37_0    defaults
sphinx                    2.4.0                      py_0    defaults
sphinxcontrib-applehelp   1.0.1                      py_0    defaults
sphinxcontrib-devhelp     1.0.1                      py_0    defaults
sphinxcontrib-htmlhelp    1.0.2                      py_0    defaults
sphinxcontrib-jsmath      1.0.1                      py_0    defaults
sphinxcontrib-qthelp      1.0.2                      py_0    defaults
sphinxcontrib-serializinghtml 1.1.3                      py_0    defaults
sqlite                    3.31.1               ha441bb4_0    defaults
tabulate                  0.8.3                    py37_0    defaults
termcolor                 1.1.0                    py37_1    defaults
tk                        8.6.8                ha441bb4_0    defaults
toml                      0.10.0           py37h28b3542_0    defaults
tqdm                      4.42.1                     py_0    defaults
traitlets                 4.3.3                    py37_0    defaults
twine                     2.0.0                      py_0    defaults
typed-ast                 1.4.1            py37h1de35cc_0    defaults
typing_extensions         3.7.4.1                  py37_0    defaults
untokenize                0.1.1                      py_1    http://bobconda.lab.idiap.ch/public/conda
urllib3                   1.25.8                   py37_0    defaults
webencodings              0.5.1                    py37_1    defaults
wheel                     0.34.2                   py37_0    defaults
xz                        5.2.4                h1de35cc_4    defaults
yaml                      0.1.7                hc338f04_2    defaults
zipp                      0.3.3            py37h8c28fb6_0    https://www.idiap.ch/software/bob/conda
zlib                      1.2.11               h1de35cc_3    defaults
zstd                      1.3.7                h5bba6e5_0    defaults
```

</details>

### PR DESCRIPTION
**Pull request by anjos**
Closes ContinuumIO/anaconda-issues#11672

### CODE REVIEW

**Comment by anjos** [comment]
@jjhelmus: thx in advance for checking this one.

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -1,9 +1,11 @@
 # cuda 9.2, gcc 7.3
 # cuda 10.0, gcc 7.3
 c_compiler_version:
-  - 7.3
+  - 7.3    # [not osx]
+  - 4.0.1  # [osx]
```
Comment (katietz) [inline recipe/conda_build_config.yaml]: We should use here instead clang version 10.0.0, or is there a strong reason to use instead old 4.0.1?

File: recipe/conda_build_config.yaml [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
 cxx_compiler_version:
-  - 7.3
+  - 7.3    # [not osx]
+  - 4.0.1  # [osx]
```
Comment (katietz) [inline recipe/conda_build_config.yaml]: Likewise

Comment (anjos) [reply recipe/conda_build_config.yaml]: Updated with 9b8e8cf

File: recipe/conda_build_config.yaml (hunk 1 above)
Comment (anjos) [reply recipe/conda_build_config.yaml]: Updated with 9b8e8cf

**Comment by anjos** [comment]
@katietz: I just updated the recipe as suggested.

**Comment by anjos** [comment]
Meanwhile, I noticed you only enabled python=3.7 builds.  I also enabled python=3.6 builds in this PR.

**Comment by anjos** [comment]
@katietz: any news about this one?

**Pull request merged**